### PR TITLE
linkedList starter

### PR DESCRIPTION
**Pull request by sahibak**
Basic starter code for linkedList

### CODE REVIEW

File: library/linkedList.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+class Node {
+  constructor() {
+    this.data = 1;
+    this.next = null;
+  }
+
+  addLink() {
+    let nextNode = this.next;
+    if (nextNode === null) {
```
Comment (yagnik) [inline library/linkedList.js:11]: you can remove this if, the while is enough

Comment (sahibak) [reply library/linkedList.js]: removed

File: library/linkedList.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+     * adding a node to the begging of the chain
+     * eg. intial node => Node(data:1, next:(data:2, next:null)) and value = "xyz"
+     * new node => Node(data:"xyz", next:(data:1, next:(data:2, next:null)))
+     * to achive the above, values of this.data and this.next is saved in the newNode => making newNode mimic this.
+     * then at this.data inserting value param and this.next's value to be the new node.
+     * @param {any} value
+     * @return {node}
+     */
+    let newNode = new Node();
+    newNode.data = this.data;
```
Comment (yagnik) [inline library/linkedList.js:43]: can be shortened with `newNode = new Node(this.data)`

File: library/linkedList.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+     */
+    if (!Number.isInteger(positionToInsert)) {
+      throw new Error("Provide integer value for position.");
+    }
+    if (positionToInsert === 0) {
+      this.addLinkToFront(value);
+    }
+    let nextNode = this;
+    let currentPosition = 1;
+    while (nextNode.next != null && currentPosition != positionToInsert) {
```
Comment (yagnik) [inline library/linkedList.js:66]: notice the duplication of some code between here and add to back,
can we re-use some pieces ?

File: library/linkedList.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+  removeFirst() {
+    /**
+     * removing the first node in the chain
+     * eg. intial node => Node(data:1, next:(data:2, next:null))
+     * new node => Node(data:2,next:null)
+     * if there are no links, it will make the class values of keys to be null
+     * otherwise, it will remove the first link and set the second link as first.
+     * @return {node}
+     */
+    let newNodes = this.next;
```
Comment (yagnik) [inline library/linkedList.js:87]: nodes ? plural 💅

Comment (sahibak) [reply library/linkedList.js:87]: resolved.

File: library/linkedList.js (hunk 2 above)
Comment (sahibak) [reply library/linkedList.js:43]: Resolved.

File: library/linkedList.js (hunk 3 above)
Comment (sahibak) [reply library/linkedList.js:66]: I don't think i can re-use as i don't have currentPosition and positionToInsert variables under addToBack. I can add currentPosition to addToBack, but PositionToInsert will still be missing. If you are not satisfied with my reasoning, please let me know and I will look for alternate solutions to reuse addToBack under addLink(value, positionToInsert) method.

**Pull request merged**